src/sqlfluff/rules/L049.py

```diff
@@ -35,85 +39,85 @@ class Rule_L049(Rule_L006):
         WHERE a IS NULL
     """
 
-    def _eval(self, context: RuleContext) -> LintResult:
+    def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
         """Relational operators should not be used to check for NULL values."""
         # Context/motivation for this rule:
         # https://news.ycombinator.com/item?id=28772289
         # https://stackoverflow.com/questions/9581745/sql-is-null-and-null
         if len(context.segment.segments) <= 2:
-            return LintResult()
+            return None
 
         # Allow assignments in SET clauses
         if context.parent_stack and context.parent_stack[-1].is_type(
             "set_clause_list", "execute_script_statement"
         ):
-            return LintResult()
+            return None
 
         # Allow assignments in EXEC clauses
         if context.segment.is_type("set_clause_list", "execute_script_statement"):
-            return LintResult()
+            return None
 
+        segment = context.functional.segment
         # Iterate through children of this segment looking for equals or "not
         # equals". Once found, check if the next code segment is a NULL literal.
-        idx_operator = None
-        operator = None
-        for idx, sub_seg in enumerate(context.segment.segments):
-            # Skip anything which is whitespace or non-code.
-            if sub_seg.is_whitespace or not sub_seg.is_code:
+
+        children = segment.children()
+        operators = segment.children(sp.is_name("equals", "not_equal_to"))
+        if len(operators) == 0:
+            return None
+
+        results: List[LintResult] = []
+        # We may have many operators
+        for operator in operators:
+            sub_list = children.select(start_seg=operator)
```

Comment:
Could we rename `sub_list` as `after_operator` or something like that? I think it'd be a bit more descriptive and readable.

Reply:
Fixed